omod/src/main/webapp/resources/scripts/taggedObjects.js

```diff
@@ -0,0 +1,41 @@
+
+var app = angular.module('taggedObjects',['personService','tagService']);
+
+app.config(['$locationProvider', function($locationProvider){
+    $locationProvider.html5Mode({
+        enabled: true,
+        requireBase: false
+    });
+}]);
+
+app.controller('taggedObjectsCtrl',['$scope','$location','Person','TagService',
+    function ($scope,$location,Person,TagService) {
+
+    var tags = [];
+    $scope.tag = $location.search().tag;
+    $scope.dataLoaded = false;
+    $scope.loadTags =function () {
+        var payload = {
+            objectType: 'org.openmrs.Patient',
+            tag: $scope.tag,
+            v: 'full'
+        };
+        TagService.getTags(payload).then(function (res) {
+            tags = res;
+            loadPatients();
+        });
+    };
+
+    $scope.patients = [];
+    var loadPatients = function () {
+        angular.forEach(tags, function (t) {
+            var payload = {
+                uuid: t.objectUuid,
+            }
+           Person.get(payload).$promise.then(function (res) {
+               $scope.patients.push(res);
+           });
+        });
+        $scope.dataLoaded = true;
```

Comment:
This still won't work,  you are making asynchronous calls, it implies this will be called after all the requests are dispatched and not when they all return.